README.md

```diff
@@ -1,31 +1,784 @@
+
+#### Implementations
+
+*NoErrorHandler*
+
+`NoErrorHandler` does nothing (`canResolveError` always returns `false`).
+
+*NoInternetErrorHandler*
+
+```Swift
+final class NoInternetErrorHandler: BaseStatusCodeErrorHandler {
+
+    init(maxRepetitions: Int = 3, delay: DispatchTimeInterval = .seconds(1))
+}
+```
+
+`NoInternetErrorHandler` handles the status code `599` by calling `retry` (see [retry](#retry)). Parameters for `retry` can be set in `init`.
+
+*RequestTimeOutErrorHandler*
+
+```Swift
+final class RequestTimeOutErrorHandler: BaseStatusCodeErrorHandler {
+
+    init(maxRepetitions: Int = 3, delay: DispatchTimeInterval = .seconds(0))
+}
+```
+
+`RequestTimeOutErrorHandler` handles the status code `408` by calling `retry` (see [retry](#retry)). Parameters for `retry` can be set in `init`.
+
+### RxFetcher
+
+Instead of using the callbacks, you can make your requests in a more "reactive" way by using `Observable`. To do that, you need an instance of `RxFetcher`, which can be obtained by `rx` property of `Fetcher`. `RxFetcher` has the same `request` methods, but they return `Observable`. Modified third example from [Quick overview](#quick-overview):
+
+```Swift
+let fetcher = Fetcher(requestPerformer: AlamofireRequestPerformer())
+fetcher.register(requestEnhancers: HttpBinResponseTranslation())
+fetcher.register(requestModifiers: BaseUrl(baseUrl: "https://httpbin.org"))
+
+fetcher.rx.request(Endpoints.get(id: "1", text: "a").subscribe(onNext: { response in
+    switch response.result {
+    case .success(let value):
+        // value was already transformed with DataMapper.
+        print(value)
+    case .failure(let error):
+        // Handle error
+        break
+    }
+}).addDisposableTo(...)
+```
+
+#### Observable extensions
+
+```Swift
+extension ObservableConvertibleType where E: ResponseProtocol {
+
+    func retryRequest(max: Int = Int.max, delay: DispatchTimeInterval = .seconds(0)) -> Observable<E>
+    
+    func asResult() -> Observable<FetcherResult<E.T>>
+}
+```
+
+`retryRequest` is a reactive variant to `retry` of `Request` (see [retry](#retry)). 
+
+`asResult` returns a mapped sequence with `Response.result`.
+
+### Thread safety
+
+`Fetcher` by default does many things on the background thread, so it needs to be thread safe. This also applies to things like `RequestPerformer`, `RequestEnhancer`, `ErrorHandler` and so on. Usually all methods from these protocols are  implemented as pure functions, so they don't cause any problems. Also see [DataMapper#ThreadSafety](https://github.com/Brightify/DataMapper#threadsafety). 
+
+## Versioning
+
+This library uses semantic versioning. Until the version 1.0 API breaking changes may occur even in minor versions. We consider the version 0.1 to be prerelease, which means that API should be stable but is not tested yet in a real project. After that testing, we make needed adjustments and bump the version to 1.0 (first release).
```

Comment:
"After the testing we'll make the necessary adjustments and bump the version to 1.0 (first release)."